### ISSUE TITLE
Expand listener priorities

### ISSUE THREAD

**Issue by SamB440** [retrooper/packetevents#911]
**Is your feature request related to a problem? Please describe.**
Packet listeners are currently limited by an enum: https://github.com/retrooper/packetevents/blob/2.0/api/src/main/java/com/github/retrooper/packetevents/event/PacketListenerPriority.java

When not shading and multiple plugins are using packetevents, this can become problematic, if two plugins listen on a priority then the order in which they execute is unknown, and plugins such as anticheats would like to be after any packet changes made by plugins.

**Describe the solution you'd like**
The addition of priority integers instead of an enum.

**Describe alternatives you've considered to solve your solution without
us adding this as a feature?**
Shading to stay ahead of plugins using the packetevents plugin instance.

**Additional context**

**Comment by retrooper** [comment]
You mention that AntiCheats, such as yours, would like to be *after* packet changes by other plugins. Then at the end you mention that you currently "shade" to stay ahead of all plugins. (implying before) Which is it now?

**Comment by SamB440** [comment]
Currently shading to be after the packet changes by other plugins.

**Comment by retrooper** [comment]
You're aware that this does not solve the packet-modification of other plugins that shade PacketEvents, right? If two plugins just shade PacketEvents, the order in which the *netty* handlers are invoked is not predictable and depends on whichever plugin loads first.\
Moreover, you could just opt for a later priority in Grim to avoid packet changes. Reputable plugins shouldn't be modifying packets without allowing others to account for it.

**Comment by SamB440** [comment]
> You're aware that this does not solve the packet-modification of other plugins that shade PacketEvents, right? If two plugins just shade PacketEvents, the order in which the netty handlers are invoked is not predictable and depends on whichever plugin loads first.

Yes, this is why I have all my other plugins use the plugin packetevents, but the anticheat shaded.

> Moreover, you could just opt for a later priority in Grim to avoid packet changes.

Grim already uses all packet priorities in the enum, I can't just simply make them all later, but I can with int priorities.
